geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/CentroidManagerGeoWave.java

```diff
@@ -0,0 +1,669 @@
+import mil.nga.giat.geowave.analytics.tools.RunnerUtils;
+import mil.nga.giat.geowave.analytics.tools.SimpleFeatureItemWrapperFactory;
+import mil.nga.giat.geowave.index.ByteArrayId;
+import mil.nga.giat.geowave.index.StringUtils;
+import mil.nga.giat.geowave.store.CloseableIterator;
+import mil.nga.giat.geowave.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.store.adapter.WritableDataAdapter;
+import mil.nga.giat.geowave.store.index.Index;
+import mil.nga.giat.geowave.store.index.IndexType;
+import mil.nga.giat.geowave.vector.adapter.FeatureDataAdapter;
+import mil.nga.giat.geowave.vector.query.AccumuloCqlConstraintsQuery;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.commons.cli.Option;
+import org.apache.commons.collections.map.LRUMap;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.conf.Configuration;
+import org.geotools.filter.FilterFactoryImpl;
+import org.opengis.filter.Filter;
+import org.opengis.filter.expression.Expression;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+
+/**
+ * 
+ * Manages the population of centroids by group id and batch id.
+ * 
+ * Properties:
+ * 
+ * @formatter:off
+ * 
+ *                "CentroidManagerGeoWave.Centroid.WrapperFactoryClass" -
+ *                {@link AnalyticItemWrapperFactory} to extract wrap spatial
+ *                objects with Centroid management function
+ * 
+ *                "CentroidManagerGeoWave.Centroid.DataTypeId" -> The data type
+ *                ID of the centroid simple feature
+ * 
+ *                "CentroidManagerGeoWave.Centroid.IndexId" -> The GeoWave index
+ *                ID of the centroid simple feature
+ * 
+ *                "CentroidManagerGeoWave.Global.BatchId" -> Batch ID for
+ *                updates
+ * 
+ *                "CentroidManagerGeoWave.Global.Zookeeper" -> Zookeeper URL
+ * 
+ *                "CentroidManagerGeoWave.Global.AccumuloInstance" -> Accumulo
+ *                Instance Name
+ * 
+ *                "CentroidManagerGeoWave.Global.AccumuloUser" -> Accumulo User
+ *                name
+ * 
+ *                "CentroidManagerGeoWave.Global.AccumuloPassword" -> Accumulo
+ *                Password
+ * 
+ *                "CentroidManagerGeoWave.Global.AccumuloNamespace" -> Accumulo
+ *                Table Namespace
+ * 
+ *                "CentroidManagerGeoWave.Common.AccumuloConnectFactory" ->
+ *                {@link BasicAccumuloOperationsBuilder}
+ * 
+ * @formatter:on
+ * 
+ * @param <T>
+ *            The item type used to represent a centroid.
+ */
+public class CentroidManagerGeoWave<T> implements
+		CentroidManager<T>
+{
+	final static Logger LOGGER = LoggerFactory.getLogger(CentroidManagerGeoWave.class);
+
+	private final String centroidDataTypeId;
+	private final String indexId;
```

Comment:
Each constructor grabs the index from the index store, but then most uses of the index look it up again by index ID.  Is it reasonable to just use the member variable 'index' everywhere and to drop the indexStore and indexId as member variables?